doc/nvim-tree.lua-float-preview.txt

```diff
@@ -0,0 +1,134 @@
+the form of a wrapper around |nvim-tree-api| calls in your |nvim-tree.on_attach|
+function. See |nvim-tree-mappings| for more details.
+
+Common functions to wrap may include:
+- |nvim-tree-api.node| for opening and previewing nodes
+- |nvim-tree-api.tree| to handle tree open/close
+- |nvim-tree-api.fs| for file creation
+
+A new???? (JMarkin TODO) instance of float preview must? be created: >
+  local prev, float_close_wrap = require("float-preview"):new()
+<
+The float preview is then attached to the nvim-tree buffer: >
+  prev:attach(bufnr)
+<
+The `float_close_wrap` function is then used to wrap |nvim-tree-api calls e.g.: >
+  float_close_wrap(api.node.open.edit)
+<
+Full example |nvim-tree.on_attach|: >
+  local function my_on_attach(bufnr)
+    local api = require("nvim-tree.api")
+    local FloatPreview = require("float-preview")
+
+    local function opts(desc)
+      return { desc = "nvim-tree: " .. desc, buffer = bufnr, noremap = true, silent = true, nowait = true }
+    end
+
+    local prev, float_close_wrap = FloatPreview:new()
+
+    vim.keymap.set("n", "<C-t>", float_close_wrap(api.node.open.tab), opts("Open: New Tab"))
+    vim.keymap.set("n", "<C-v>", float_close_wrap(api.node.open.vertical), opts("Open: Vertical Split"))
+    vim.keymap.set("n", "<C-s>", float_close_wrap(api.node.open.horizontal), opts("Open: Horizontal Split"))
+    vim.keymap.set("n", "<CR>", float_close_wrap(api.node.open.edit), opts("Open"))
+    vim.keymap.set("n", "<Tab>", float_close_wrap(api.node.open.preview), opts("Preview"))
+    vim.keymap.set("n", "o", float_close_wrap(api.node.open.edit), opts("Open"))
+    vim.keymap.set("n", "O", float_close_wrap(api.node.open.no_window_picker), opts("Open: No Window Picker"))
+    vim.keymap.set("n", "q", float_close_wrap(api.tree.close), opts("Close"))
+    vim.keymap.set("n", "a", float_close_wrap(api.fs.create), opts("Create"))
+    vim.keymap.set("n", "d", float_close_wrap(api.fs.remove), opts("Delete"))
+    vim.keymap.set("n", "r", float_close_wrap(api.fs.rename), opts("Rename"))
+  end
+<
+Do not forget to set `my_on_attach` in |nvim-tree-setup| e.g. >
+  require("nvim-tree").setup {
+    ---
+    on_attach = my_on_attach,
+    ---
+  }
+<
+==============================================================================
+ 4. API                                                    *float-preview-api*
+
+is_float({bufnr})                                *float-preview-api.is_float()*
+    Open the tree, focusing it if already open.
+
+    Parameters: ~
+      • {bufnr} (number|nil) buffer handle, 0 or nil for current buffer
+
+    Return: ~
+      (boolean) buffer is a floating preview
+
+TODO JMarkin I am not sure what else you want to make public. :new ?
+
+==============================================================================
+ 5. DIAGNOSTICS                                  *float-preview-diagnostics*
+
+TODO JMarkin I am not explaining why this is necessary
```

Comment:
In my usages if I enabled nullls or other external diagnostic plugin, it attached to buffer automatically and it loads system. I think many users want to recognize what it is float buffer from nvim-tree

Reply:
That sounds reasonable.